internal/fields/validate.go

```diff
@@ -345,18 +370,109 @@ func (v *Validator) validateScalarElement(key string, val interface{}, doc commo
 		val = fmt.Sprintf("%q", val)
 	}
 
-	err := v.validateExpectedNormalization(*definition, val)
-	if err != nil {
-		return errors.Wrapf(err, "field %q is not normalized as expected", key)
+	if !v.disabledNormalization {
+		err := v.validateExpectedNormalization(*definition, val)
+		if err != nil {
+			return fmt.Errorf("field %q is not normalized as expected: %w", key, err)
+		}
 	}
 
-	err = v.parseElementValue(key, *definition, val, doc)
+	err := v.parseElementValue(key, *definition, val, doc)
 	if err != nil {
-		return errors.Wrap(err, "parsing field value failed")
+		return fmt.Errorf("parsing field value failed: %w", err)
 	}
 	return nil
 }
 
+func (v *Validator) SanitizeSyntheticSourceDocs(docs []common.MapStr) ([]common.MapStr, error) {
+	var newDocs []common.MapStr
+	for _, doc := range docs {
+		for key, contents := range doc {
+			definition := FindElementDefinition(key, v.Schema)
+			if definition == nil {
+				continue
+			}
+
+			if !v.disabledDependencyManagement && definition.External != "" {
+				def, err := v.FieldDependencyManager.ImportField(definition.External, key)
+				if err != nil {
+					return nil, fmt.Errorf("can't import field (field: %s): %w", key, err)
+				}
+				definition = &def
+			}
+
+			shouldBeArray := false
+			// normalization should just be checked if synthetic source is enabled and the
+			// spec version of this package is >= 2.0.0
+			if v.disabledNormalization && !v.specVersion.LessThan(semver2_0_0) {
+				for _, normalize := range definition.Normalize {
+					switch normalize {
+					case "array":
+						shouldBeArray = true
+					}
+					if shouldBeArray {
+						break
+					}
+				}
+			}
```

Comment:
Maybe something like this to prefer single values over arrays for fields without definition.
```suggestion
			shouldBeArray := false
			if definition != nil {
				if !v.disabledDependencyManagement && definition.External != "" {
					def, err := v.FieldDependencyManager.ImportField(definition.External, key)
					if err != nil {
						return nil, fmt.Errorf("can't import field (field: %s): %w", key, err)
					}
					definition = &def
				}

				// normalization should just be checked if synthetic source is enabled and the
				// spec version of this package is >= 2.0.0
				if v.disabledNormalization && !v.specVersion.LessThan(semver2_0_0) {
					for _, normalize := range definition.Normalize {
						switch normalize {
						case "array":
							shouldBeArray = true
						}
						if shouldBeArray {
							break
						}
					}
				}
			}
```

Reply:
Nit. And consider moving the code to check if it should be an array to its own function.